src/kernel/src/process/rlimit.rs

```diff
@@ -1,4 +1,75 @@
-use std::io::Error;
+use std::{io::Error, ops::Index};
+use thiserror::Error;
+
+//TODO: add remaining limits
+#[derive(Debug, Clone, Copy)]
+pub enum ResourceType {
+    Cpu = 0,
+    Fsize = 1,
+    Data = 2,
+}
+
+impl ResourceType {
+    #[cfg(target_os = "linux")]
+    pub fn into_unix(self) -> libc::__rlimit_resource_t {
```

Comment:
`into_unix` is quite confused because PS4 is also *nix.

Reply:
So... 'into_host'?